### PR TITLE
Added an XML reference

### PR DESCRIPTION
**Pull request by dantleech**
Added simple XML mapping chapter that referrers back to the annotation reference for details.

### CODE REVIEW

File: en/reference/xml-reference.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+XML Mapping Reference
+=====================
+
+This chapter gives a brief overview of the XML mapping by example. In general
```
Comment (dbu) [inline en/reference/xml-reference.rst]: , after "In general"?

File: en/index.rst [hunk 2]
```diff
@@ -33,9 +33,9 @@ Mapping Objects onto a Document Repository
   :doc:`Hierarchies and References <reference/association-mapping>` |
   :doc:`Inheritance <reference/inheritance-mapping>`
 
-* **Mapping Driver References**: (TODO: build this in a sane way, without duplicating too much information)
+* **Mapping Driver References**:
   :doc:`Docblock Annotations <reference/annotations-reference>` |
-  :doc:`XML <reference/xml-mapping>` |
+  :doc:`XML <reference/xml-reference>` |
```
Comment (dbu) [inline en/index.rst]: i think calling this xml-mapping would be more correct, as it is not a reference.

**Comment by dbu** [comment]
thanks for picking this up! i have two details, otherwise lets merge this.

File: en/reference/xml-reference.rst (hunk 1 above)
Comment (dantleech) [inline en/reference/xml-reference.rst]: I see it as a single phrase and when speaking it there would be no pause that is associated with a comma. Though tbh I am not sure what the exact gramatical rules are here.

**Comment by dantleech** [comment]
Updated, added the comma as I don't think it makes a difference either way.

**Comment by dantleech** [comment]
Also renamed `annotation-reference` to `annotation-mapping`.

**Comment by dbu** [comment]
thanks dan!

**Pull request merged**